src/actionlib/ActionListener.ts

```diff
@@ -17,42 +19,60 @@ import { EventEmitter } from "eventemitter3";
  *
  *
  */
-export default class ActionListener extends EventEmitter {
-  ros;
-  serverName;
-  actionName;
+export default class ActionListener<
+  TGoal,
+  TFeedback,
+  TResult,
+> extends EventEmitter {
+  ros: Ros;
+  serverName: string;
+  actionName: string;
   /**
-   * @param {Object} options
-   * @param {import('../core/Ros.js').default} options.ros - The ROSLIB.Ros connection handle.
-   * @param {string} options.serverName - The action server name, like '/fibonacci'.
-   * @param {string} options.actionName - The action message name, like 'actionlib_tutorials/FibonacciAction'.
+   * @param options
+   * @param options.ros - The ROSLIB.Ros connection handle.
+   * @param options.serverName - The action server name, like '/fibonacci'.
+   * @param options.actionName - The action message name, like 'actionlib_tutorials/FibonacciAction'.
    */
-  constructor({ ros, serverName, actionName }) {
+  constructor({
+    ros,
+    serverName,
+    actionName,
+  }: {
+    ros: Ros;
+    serverName: string;
+    actionName: string;
+  }) {
     super();
     this.ros = ros;
     this.serverName = serverName;
     this.actionName = actionName;
 
     // create the topics associated with actionlib
-    const goalListener = new Topic({
+    const goalListener = new Topic<TGoal>({
       ros: this.ros,
       name: this.serverName + "/goal",
       messageType: this.actionName + "Goal",
     });
 
-    const feedbackListener = new Topic({
+    const feedbackListener = new Topic<{
+      status: actionlib_msgs.GoalStatus;
```

Comment:
is `GoalStatus` means to be in the feedback listener?

Reply:
I, too, am baffled by this. But the code does refer to it.

Reply:
ooh there actually may be something more sinister going on in there.. this might be a _different_ "status"